### PR TITLE
Prep for Docker log drivers announcement

### PR DESCRIPTION
**Pull request by tgross**
@misterbisson I'm opening this PR for review, but I'll want to add some more content for the README before merging. This will complement what'll probably be more extensive work in `triton-elk` that I'll be working on through early next week. The current master branch prior to merging this was tagged as `preview`.

Changes:
- switch to Alpine Linux
- install ES from the source tarball because there's no APK, which required a lot of fiddly Dockerfile
- check package signatures for ES and Containerbuddy during build
- discover Consul via Triton CNS when deployed to Triton (but works locally via envvar)
- consolidate the Docker Compose files to take advantage of getting rid of `--link`s in the main file

### CODE REVIEW

**Comment by tgross** [comment]
Another note is that I considered trying to adapt the primary-replica bootstrap that we did for `triton-mysql` but there are a couple complications to that:
- ES doesn't have a simple primary-replica setup; there can be multiple masters, data-only nodes, and nodes that serve as both masters and data. How you divide that up is highly idiosyncratic to the workload you have.
- given how long `triton-mysql` took, it's probably more work than we probably want to invest in this blueprint right now. :grinning:

**Comment by misterbisson** [comment]
> given how long `triton-mysql` took

Agreed.

> ES doesn't have a simple primary-replica setup

Well, this is relatively easy if we can eliminate the hybrid master-data nodes, no?

``` yml
es-master:
    image: elasticsearch
    command: do-master

es-data:
    image: elasticsearch
    command: do-data
```

They're both the same image, but a different command to start. 

On starting an existing master/cluster, it will look for an existing master, and if it finds one, join it as a peer. This is similar to our Couchbase work. If there is no existing cluster, it will form one. So, the masters are an upstream for each other.

On starting a data node, they also look for an existing cluster, but fail if there is none. They, too, have the masters as upstreams.

OTOH, this also easy if we break that single Compose file into three. They can still have the same behaviors as described above (hybrid nodes would mostly behave like masters, I guess), but you wouldn't be forced to start one of each at first. The big difference, I guess, is that hybrid nodes would announce both the master and data services, no?

**I'm probably wrong about all of that, however, and I stand by the point at the top: out of scope.**

File: Dockerfile [hunk 1]
```diff
@@ -1,42 +1,51 @@
-FROM debian:jessie
+FROM alpine:3.3
```
Comment (misterbisson) [inline Dockerfile]: Depends on `OS-5184`, as discussed elsewhere. We should ticket a separate issue to update this when `OS-5184` is shipped.

**Comment by tgross** [comment]
@misterbisson this and https://github.com/tgross/triton-elk/pull/2 are probably ready for re-review at this point.

File: Dockerfile [hunk 2]
```diff
@@ -10,21 +10,23 @@ RUN apt-get update && \
...
+ENV CONTAINERBUDDY_VERSION 1.2.1
+RUN export CB_SHA1=aca04b3c6d6ed66294241211237012a23f8b4f20 \
+    && curl -Lso /tmp/containerbuddy.tar.gz \
+        "https://github.com/joyent/containerbuddy/releases/download/${CONTAINERBUDDY_VERSION}/containerbuddy-${CONTAINERBUDDY_VERSION}.tar.gz" \
+    && echo "${CB_SHA1}  /tmp/containerbuddy.tar.gz" | sha1sum -c \
+    && tar zxf /tmp/containerbuddy.tar.gz -C /bin \
+    && rm /tmp/containerbuddy.tar.gz
+
+# Create and take ownership over required directories
 # Take ownership over required directories
```
Comment (misterbisson) [inline Dockerfile]: Dupe comment?

**Comment by misterbisson** [comment]
Solid. Small comment, but :house_with_garden: :walking:.

File: Dockerfile (hunk 2 above)
Comment (tgross) [inline Dockerfile]: Copy-pasta bug! Fixed!

**Pull request merged**